### DISCUSSION TITLE
Testing HTTP versions offered by endpoint

### DISCUSSION THREAD

**Discussion by ragebflame** [Orange-OpenSource/hurl#1705] [Q&A] [answered]
Hi,

I am looking for a way of testing if my API is serving content over the various HTTP versions. I have things configured to only server HTTP/2, but I would like a way of confirming this. 

_sample.hurl_
```
GET https://media-laptop.lan
HTTP/1.0 200

GET https://media-laptop.lan
HTTP/1.1 200

GET https://media-laptop.lan
HTTP/2 200
```

_hurl -k --test sample.hurl_
```
sample.hurl: Running [1/1]
error: Assert HTTP version
  --> sample.hurl:2:1
   |
 2 | HTTP/1.0 200
   | ^^^^^^^^ actual value is <HTTP/2>
   |

sample.hurl: Failure (1 request(s) in 57 ms)
--------------------------------------------------------------------------------
Executed files:  1
Succeeded files: 0 (0.0%)
Failed files:    1 (100.0%)
Duration:        57 ms
```

I have been reading the documentation around [HTTP version responses](https://hurl.dev/docs/asserting-response.html#version-status), but don't see a way of asserting on the value returned.

**Comment by jcamiel** [comment]
Hi @ragebflame if I understand correctly, you would like a way to test that your version is not `HTTP/1.0` or not `HTTP/1.1` ?
Currently, we have implicit and explicit assert on status:

Implicit:

```hurl
GET https://foo.com
HTTP 200
```

Explicit:

```hurl
GET https://foo.com
HTTP *
[Asserts]
status == 200
```

We have implicit assert on version as well:

```hurl
GET https://foo.com
HTTP/2 200
```

But no explicit assert on version. It makes sense to be able to write:

```hurl
GET https://foo.com
HTTP 200
[Asserts]
version != HTTP/1.0
```

If this is what you want, we can turn this discussion to an issue for our TODO list

**Comment by ragebflame** [comment]
Hi @jcamiel,

Thanks for the quick response.

> Hi @ragebflame if I understand correctly, you would like a way to test that your version is not `HTTP/1.0` or not `HTTP/1.1` ? Currently, we have implicit and explicit assert on status:

Yes, this is what I was looking for. A way to ensure that the endpoint is not serving requests over a HTTP version lower than 2.

> But no explicit assert on version. It makes sense to be able to write:
> 
> ```
> GET https://foo.com
> HTTP 200
> [Asserts]
> version != HTTP/1.0
> ```
> 
> If this is what you want, we can turn this discussion to an issue for our TODO list

This is what I meant (but worded things poorly!). I'd be happy for this to be added as a TODO item if you guys think it would be a useful addition.

**Comment by jcamiel** [comment, marked as answer]
Issue created here => #1706

**Reply by ragebflame** [reply]
Great. Thanks!

**Reply by jcamiel** [reply]
Hi @ragebflame 

The `version` explicit query has been implemented on the just released Hurl 6.1.0!

**Reply by ragebflame** [reply]
Thanks @jcamiel 

This will be very useful for what I was trying to do. Appreciate it.
